packages/scandipwa/src/component/Fallback/Fallback.component.tsx

```diff
@@ -0,0 +1,161 @@
+                <p>
+                    <TextPlaceholder length={ TextPlaceHolderLength.LONG } />
+                </p>
+            );
+        }
+
+        if (!description) {
+            return null;
+        }
+
+        return (
+            <div block="Fallback" elem="CategoryDescription">
+                { parser(description) }
+            </div>
+        );
+    }
+
+    renderCategoryFallback(): ReactElement {
+        const { actionName: { type } = {} } = window;
+
+        if (type !== UrlRewritePageType.CATEGORY) {
+            return this.renderDefaultFallback();
+        }
+
+        return (
+            <main
+              block="Fallback"
+              elem="CategoryPagePlaceholder"
+            >
+                <ContentWrapper
+                  wrapperMix={ { block: 'Fallback', elem: 'CategoryWrapper' } }
+                  label="Category page"
+                >
+                    <div
+                      block="Fallback"
+                      elem="CategoryFilterOverlay"
+                      mix={ { block: 'Fallback', elem: 'Placeholder' } }
+                    />
+                    <article block="Fallback" elem="CategoryDetails">
+                        { this.renderCategoryName() }
+                        { this.renderCategoryDescription() }
+                    </article>
+                    <div block="Fallback" elem="DefaultPlaceholder" />
+                </ContentWrapper>
+            </main>
+        );
+    }
+
+    renderProductFallback(): ReactElement {
+        const { actionName: { name, type } = {} } = window;
+
+        if (type !== UrlRewritePageType.PRODUCT) {
+            return this.renderDefaultFallback();
+        }
+
+        return (
+            <main
+              block="Fallback"
+              elem="ProductPagePlaceholder"
+              aria-label="Product page"
+            >
+                <ContentWrapper
+                  wrapperMix={ { block: 'Fallback', elem: 'ProductWrapper' } }
+                  label={ __('Main product details') }
+                >
+                    <div block="Fallback" elem="ProductDetails">
+                        <h1 block="Fallback" elem="Heading">
+                            { name }
+                        </h1>
+                    </div>
+                    <div block="Fallback" elem="ProductGallery">
+                        <div block="Fallback" elem="ProductGallerySlider" />
+                    </div>
+                    <article block="Fallback" elem="ProductActions">
+                        <h1 block="Fallback" elem="Title">
+                            { name }
+                        </h1>
+                    </article>
+                </ContentWrapper>
+            </main>
+        );
+    }
+
+    renderDefaultFallback(): ReactElement {
+        return (
+            <main block="Fallback" elem="DefaultPlaceholder" />
+        );
+    }
+
+    render(): ReactElement {
+        const { type } = this.props;
+
```

Comment:
from window type. Otherwise it repeats on line 85, 116

if (!Object.keys(this.fallbackMap).includes(type)) {
     return this.renderDefaultFallback();
}